flink-test-utils-parent/flink-connector-testing/src/main/java/org/apache/flink/connectors/test/common/environment/MiniClusterTestEnvironment.java

```diff
@@ -128,6 +142,38 @@ private void startTaskManager() throws Exception {
         LOG.debug("New TaskManager {} has been launched.", latestTMIndex);
     }
 
+    @SuppressWarnings("unchecked")
+    private void waitUntilJobRestarted(JobClient jobClient, Deadline timeout) throws Exception {
+        checkNotNull(this.metricReporter, "In-memory metric reporter should exists");
+
+        // Get metric group of the running job
+        final Optional<JobManagerJobMetricGroup> jobMetricGroup =
+                metricReporter.findGroups("jobmanager").stream()
+                        .filter(group -> group instanceof JobManagerJobMetricGroup)
```

Comment:
ideally we avoid references to the internal metrics groups; can we not search for a pattern like `*<jobID>*numRestarts`?